### PR TITLE
Added Second Chance Caste to the Education Module for Warrior Caste Washouts

### PR DESCRIPTION
**Pull request by IllianiCBT**
Requires: #4127 & #4126

This PR adds the ability to select a 'second chance' Warrior Caste sub-caste for Warrior Caste washouts.

When Warrior Caste personnel washout of the Warrior Caste mhq will check that their course index does not match the index of the chosen second chance caste. If there is no match (and second chance is not set to `NONE`), then the aspirant washes into the chosen sub-caste.

If second chance is set to `NONE`, or the aspirant is washing out of the chosen sub-caste, they will washout into a civilian caste as normal.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/education/EducationController.java [hunk 1]
```diff
@@ -1610,6 +1677,27 @@ private static String graduationEventPicker() {
...
+    private static int getSecondChanceCasteIndex(Campaign campaign) {
+        switch (campaign.getCampaignOptions().getSecondChanceCaste()) {
+            case NONE:
+                return -1;
+            case BA:
+                return 4;
+            case INFANTRY:
+                return 5;
+            case VEHICLE:
+                return 6;
```
Comment (SJuliez) [inline MekHQ/src/mekhq/campaign/personnel/education/EducationController.java:1695]: This looks like it could be shortened by a default: return -1, right?

**Review by SJuliez — APPROVED** [review]

**Comment by IllianiCBT** [comment]
Resolved Conflicts

**Comment by SJuliez** [comment]
Yet again, conflicts :)
Also, see my comment. Can get rid of the NONE: case and end in default: return -1;

**Comment by IllianiCBT** [comment]
Addressed feedback and resolved conflicts

**Pull request merged**